lib/membrane_rtc_engine/endpoints/hls/mobile_layout_maker.ex

```diff
@@ -33,47 +46,87 @@ defmodule Membrane.RTC.Engine.Endpoint.HLS.MobileLayoutMaker do
+    transformations = get_transformations(output_caps, placement.size, 0)
+
+    {transformations, placement}
   end
 
-  defp get_track_layout(:blank, index, %{width: width, height: height}) do
-    position = {round(index / 2 * width), height - round(1 / 4 * height)}
-    display_size = {round(1 / 2 * width), round(1 / 4 * height)}
+  defp get_track_layout(:basic, index, caps, %{width: width, height: height}) do
+    output_caps = %{width: round(1 / 2 * width) - 10, height: round(1 / 4 * height) - 10}
+    position = {round(index / 2 * width) + 5, height - round(1 / 4 * height) + 5}
 
-    %VideoPlacement{
+    placement = get_placement(output_caps, caps, position, 0.3)
+    transformations = get_transformations(output_caps, placement.size, 20)
+
+    {transformations, placement}
+  end
+
+  defp video_proportion(%{width: width, height: height}), do: width / height
+
+  defp get_placement(output_caps, caps, position, z_value),
+    do: %BaseVideoPlacement{
       position: position,
-      display_size: display_size,
-      z_value: 0.2
+      size: get_display_size(output_caps, caps),
+      z_value: z_value
     }
+
+  defp get_display_size(output_caps, caps) do
+    if video_proportion(output_caps) > video_proportion(caps),
+      do: {output_caps.width, round(output_caps.width / caps.width * caps.height)},
+      else: {round(output_caps.height / caps.height * caps.width), output_caps.height}
   end
 
-  defp get_track_layout(:main, _index, %{width: width, height: height}),
-    do: %VideoPlacement{
-      position: {0, 0},
-      display_size: {width, height},
-      z_value: 0.1
+  defp get_transformations(output_caps, display_size, radius),
+    do: %VideoTransformations{
+      texture_transformations: [
+        get_cropping(output_caps, display_size),
+        get_corners_rounding(radius)
+      ]
     }
 
-  defp get_track_layout(:basic, index, %{width: width, height: height}) do
-    position = {round(index / 2 * width) + 5, height - round(1 / 4 * height) + 5}
-    display_size = {round(1 / 2 * width) - 10, round(1 / 4 * height) - 10}
+  defp get_corners_rounding(radius), do: %CornersRounding{border_radius: radius}
 
-    %VideoPlacement{
-      position: position,
-      display_size: display_size,
-      z_value: 0.3
+  defp get_cropping(output_caps, display_size),
+    do: %Cropping{
+      crop_top_left_corner: get_cropping_position(output_caps, display_size),
+      crop_size: get_cropping_size(output_caps, display_size),
+      cropped_video_position: :input_position
     }
+
+  defp get_cropping_position(output_caps, {width, height}) do
+    if output_caps.width == width,
+      do: {0.0, (height - output_caps.height) / (2 * height)},
+      else: {(width - output_caps.width) / (2 * width), 0.0}
+  end
+
+  defp get_cropping_size(output_caps, {width, height}) do
+    if output_caps.width == width,
+      do: {1.0, output_caps.height / height},
+      else: {output_caps.width / width, 1.0}
```

Comment:
Not necessarily in this PR, but I would add some comments on what operation is happening here.